exercises/acronym/acronym_test.go

```diff
@@ -0,0 +1,47 @@
+package acronym
+
+import (
+	"testing"
+)
+
+const targetTestVersion = 1
+
+type testCase struct {
+	input     string
+	expected  string
+	expectErr bool
+}
+
+var stringTestCases = []testCase{
+	{"Portable Network Graphics", "PNG", false},
+	{"HyperText Markup Language", "HTML", false},
+	{"Ruby on Rails", "ROR", false},
+	{"PHP: Hypertext Preprocessor", "PHP", false},
+	{"First In, First Out", "FIFO", false},
+	{"Complementary metal-oxide semiconductor", "CMOS", false},
+}
+
+func TestTestVersion(t *testing.T) {
+	if testVersion != targetTestVersion {
+		t.Errorf("Found testVersion = %v, want %v.", testVersion, targetTestVersion)
+	}
+}
+
+func TestAcronym(t *testing.T) {
+	for _, test := range stringTestCases {
+		actual, actualErr := abbreviate(test.input)
+		if !test.expectErr {
+			if actualErr != nil {
+				// if we don't expected error but got error
+				var _ error = actualErr
```

Comment:
After looking at a few of these error handling pull requests I've started to think we should always put this right after the place where the error gets assigned. In this case, right after line 31. @petertseng do you see any issue with that?